hypershift-operator/releaseinfo/releaseinfo.go

```diff
@@ -2,72 +2,140 @@ package releaseinfo
 
 import (
 	"bytes"
+	"context"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"regexp"
 	"sort"
 	"strings"
+	"time"
 
 	"github.com/blang/semver"
 	imageapi "github.com/openshift/api/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
-type ReleaseImageInfo struct {
-	Image      string                   `json:"image"`
-	Metadata   ReleaseImageInfoMetadata `json:"metadata"`
-	References imageapi.ImageStream     `json:"references"`
+type Provider interface {
+	Lookup(image string, ctx context.Context) (*ReleaseImage, error)
 }
 
-type ReleaseImageInfoMetadata struct {
-	Kind    string `json:"kind"`
-	Version string `json:"version"`
-}
+var _ Provider = (*PodProvider)(nil)
+
+type PodProvider struct {
+	Pods v1.PodInterface
 
-type ReleaseInfo struct {
-	Images   map[string]string
-	Versions map[string]string
+	// TODO: add an ExpirationCache
 }
 
-func (info *ReleaseImageInfo) ReleaseInfo(originReleasePrefix string) (*ReleaseInfo, error) {
-	var newImagePrefix string
-	if !strings.Contains(info.Image, originReleasePrefix) {
-		newImagePrefix = strings.Replace(info.Image, ":", "-", -1)
+func (p *PodProvider) Lookup(image string, ctx context.Context) (releaseImage *ReleaseImage, err error) {
```

Comment:
by convention, context is usually the first parameter

Reply:
Fixed